lib/mfg.c

```diff
@@ -111,34 +111,63 @@ static void RSA_get0_key(const RSA *r, const BIGNUM **n,
 }
 #endif
 
-/**
- * @brief Get secure boot configurations
- * @param[in]  dev	Switchtec device handle
- * @param[out] state	Current secure boot settings
- * @return 0 on success, error code on failure
- */
-int switchtec_security_config_get(struct switchtec_dev *dev,
-				  struct switchtec_security_cfg_state *state)
+static int secure_config_get(struct switchtec_dev *dev,
+			     struct switchtec_security_cfg_state *state,
+			     struct switchtec_security_cfg_otp_region *otp,
+			     bool *otp_valid)
 {
 	int ret;
+	uint8_t subcmd = 0;
 	struct cfg_reply {
 		uint32_t valid;
 		uint32_t rsvd1;
 		uint64_t cfg;
-		uint32_t  public_key_exponent;
-		uint8_t  rsvd2;
-		uint8_t  public_key_num;
-		uint8_t  public_key_ver;
-		uint8_t  rsvd3;
-		uint8_t  public_key[SWITCHTEC_KMSK_NUM][SWITCHTEC_KMSK_LEN];
-		uint8_t  rsvd4[32];
+		uint32_t public_key_exponent;
+		uint8_t rsvd2;
+		uint8_t public_key_num;
+		uint8_t public_key_ver;
+		uint8_t rsvd3;
+		uint8_t public_key[SWITCHTEC_KMSK_NUM][SWITCHTEC_KMSK_LEN];
```

Comment:
Nit: would be nice if the white space changes were not part of functional changes.

Reply:
Added another commit.